code/lib/preview-api/src/modules/preview-web/UrlStore.ts

```diff
@@ -58,15 +59,18 @@ const getFirstString = (v: ValueOf<qs.ParsedQs>): string | void => {
   if (Array.isArray(v)) {
     return getFirstString(v[0]);
   }
-  if (isObject(v)) {
-    return getFirstString(Object.values(v).filter(Boolean) as string[]);
+  if (isObject(v as Record<PropertyKey, unknown>)) {
+    return getFirstString(
+      Object.values(v as Record<PropertyKey, unknown>).filter(Boolean) as string[]
+    );
   }
   return undefined;
 };
 
 export const getSelectionSpecifierFromPath: () => SelectionSpecifier | null = () => {
   if (typeof document !== 'undefined') {
-    const query = qs.parse(document.location.search, { ignoreQueryPrefix: true });
+    const queryStr = document.location.search ? document.location.search.slice(1) : '';
+    const query = parse(queryStr);
```

Comment:
```suggestion
    const query = parse(document.location.search.slice(1));
```
There's no need to check for `search` to exist. Afaik it'll always exist on `location` and always be a string, even if empty. Slicing an empty string will yield an empty string, so that's safe too.